gammapy/utils/units.py

```diff
@@ -64,3 +66,54 @@ def unit_from_fits_image_hdu(header):
         unit = ""
 
     return standardise_unit(unit)
+
+def energy_str_formatting(E):
+    """
+    Format energy quantities to a string representation that is more comfortable to read
+    by swithcing to the most relevant unit (keV, MeV, GeV, TeV) and changing the float precision.
+    Parameters
+    ----------    
+    E: `~astropy.units.Quantity`
+        Quantity
+    Returns
+    -------
+    str : str
+        Returns a string with energy unit formatted        
+    """    
+    i = floor(np.log10(E.to_value(u.eV)) / 3) # a new unit every 3 decades
+    if i > 4:  # default for very large values
+        unit = u.PeV
+    else:  # get best large unit
+        unit = (u.eV, u.keV, u.MeV, u.GeV, u.TeV, u.PeV)[i]
+
+    v = E.to_value(unit)
+    i=floor(np.log10(v))
+    prec=(2,1,0)[i]
```

Comment:
A default value is needed if $i >= 3$

```suggestion
    prec = (2,1,0)[i] if i < 3 else 0
```